core/src/main/java/org/wso2/msf4j/MicroservicesRunner.java

```diff
@@ -217,16 +243,23 @@ private void checkState() {
     public void start() {
         msRegistry.getSessionManager().init();
         handleServiceLifecycleMethods();
-        transportManager.startTransports();
-        isStarted = true;
-        log.info("Microservices server started in " + (System.currentTimeMillis() - startTime) + "ms");
+        serverConnectors.forEach(serverConnector -> {
+            try {
+                serverConnector.start(Collections.emptyMap());
+                isStarted = true;
+                log.info("Microservices server started in " + (System.currentTimeMillis() - startTime) + "ms");
+            } catch (ServerConnectorException e) {
+                log.error("Error while starting the Microservices server. " + e.getMessage(), e);
+                throw new RuntimeException("Error while starting the Microservices server.", e);
```

Comment:
Few points noted here:
- It would be better to throw a specific exception without using a runtime exception unless otherwise its occurrence is not known in the standard execution path. 
- It would be better to throw the exception without logging it, as it may generate unnecessary error logs if the thrown exception is logged at the caller.